third_party/asr/asr582x/BUILD.gn

```diff
@@ -0,0 +1,382 @@
+    "CFG_CMON",
+    "CFG_MDM_VER_V21",
+    "CFG_SOFTAP_SUPPORT",
+    "CFG_SNIFFER_SUPPORT",
+    "CFG_CUS_FRAME",
+    "AT_USER_DEBUG",
+    "CFG_WF_DBG=1",
+    "__FPU_PRESENT=1",
+    "DX_CC_TEE",
+    "HASH_SHA_512_SUPPORTED",
+    "CC_HW_VERSION=0xF0",
+    "DLLI_MAX_BUFF_SIZE=0x10000",
+    "SSI_CONFIG_TRNG_MODE=0",
+    "LWIP_APP_IPERF",
+    "SYSTEM_RECOVERY",
+    "LWIP_DUALSTACK",
+    "SYSTEM_COREDUMP",
+    "CFG_REC",
+    "CFG_SNIFFER_UNICAST_SUPPORT",
+    "CFG_OS_VERSION_CHECK",
+    "PS_CLOSE_APLL",
+    "CFG_STA_AP_COEX_N",
+    "CFG_SOFTAP_SWITCH_CHAN_N",
+    "CFG_PLF_DUET",
+    "SONATA_RTOS_SUPPORT",
+    "CFG_SEG",
+    "CFG_DUET_FREERTOS",
+    "PRINTF2_SUPPORT",
+    "DUET_RF_SLEEP",
+    "CFG_DUET_BLE",
+    "CFG_RTC_INDEPENDENT",
+    "CFG_SAE",
+    "CFG_SAE_AUTH_OPT",
+    "CFG_MFP",
+    "CONFIG_IEEE80211W",
+    "WFA_CERTIFICATE_N",
+    "CFG_DUET_5822S",
+    "CFG_DUET_5822T",
+    "MBEDTLS_HW",
+    "MBEDTLS_HW_RSA",
+    "MBEDTLS_HW_ECC",
+    "MBEDTLS_HW_AES",
+    "MBEDTLS_HW_HASH",
+    "MBEDTLS_USER_CONFIG_FILE=<mbedtls_user_config.h>",
+    "CONFIG_ENABLE_ASR_LEGA_RTOS=1",
+#    "CONFIG_ENABLE_ASR_APP_MESH=1",
+    "CFG_EASY_LOG_ENABLE=1",
+    "CFG_EASY_LOG_MODULE_EN=1",
+    ]
+
+  cflags_c = [
+    "-mcpu=cortex-m4",
+    "-Os",
+    "-mfloat-abi=softfp",
+    "-mfpu=fpv4-sp-d16",
+    "-mthumb",
+    "-ffunction-sections",
+    "-fdata-sections",
+    "-g",
+    "-Wno-sign-compare",
+    "-Wno-unused-function",
+    "-Wno-pointer-sign",
+    "-Wno-type-limits",
+    "-Wall"
+   ]
+  cflags_cc = [
+    "-mcpu=cortex-m4",
+    "-Os",
+    "-mfloat-abi=softfp",
+    "-mfpu=fpv4-sp-d16",
+    "-mthumb",
+    "-ffunction-sections",
+    "-fdata-sections",
+    "-g",
+    "-Wall",
+   ]
+  asmflags = [
+    "-mcpu=cortex-m4",
+    "-mfloat-abi=softfp",
+    "-mfpu=fpv4-sp-d16",
+    "-mthumb",
+    "-ffunction-sections",
+    "-fdata-sections",
+    "-g",
+    "-Wall"
+   ]
+  ldflags = [
+    "-mcpu=cortex-m4",
+    "-mfloat-abi=softfp",
+    "-mfpu=fpv4-sp-d16",
+    "-mthumb",
+    "-ffunction-sections",
+    "-fdata-sections",
+    "-g",
+    "-Wall",
+   # "-lstdc++",
+   # "-nostartfiles",
+    "-Wl,--gc-sections",
+   ]
+
+  libs = [
+     "${chip_root}/third_party/asr/${asr_ic_family}/asr_sdk/lib/5822s/libasr_combo_v6.a",
+     "${chip_root}/third_party/asr/${asr_ic_family}/asr_sdk/lib/ble/lib_ble_mesh.a",
+     "${chip_root}/third_party/asr/${asr_ic_family}/asr_sdk/lib/security/lib_security.a",
+  ]
+
+  linker_script_flags = "${asr_sdk_build_root}/asr_sdk/demo/matter_demo/app/ld/5822t/gcc.ld"
+  ldflags += [ "-T" + rebase_path(linker_script_flags, root_build_dir) ]
+  ble_symbol_flags = "${asr_sdk_build_root}/asr_sdk/lib/ble/asr_ble_rom_symbol_v20200507204845.txt"
+  ldflags += [ "-Wl,-R" + rebase_path(ble_symbol_flags, root_build_dir) ]
+}
+
+group("asr_build") {
+  public_configs = [ ":asr_sdk_config" ]
+  public_deps = [ asr_target_project ]
+}
+
+static_library("asr_mbedtls") {
```

Comment:
any reason to prefix everything with `asr` here? The folder is already in asr, so targets will be `.../ars/asr582x:asr_mbedtls` and that stutters.